soroban-env-common/src/symbol.rs

```diff
@@ -55,20 +53,78 @@ sa::const_assert!(CODE_BITS * MAX_CHARS == BODY_BITS);
 /// [RawVal]'s body as a 60-bit small "string-like" object, 10 characters or
 /// less and with characters drawn from the 64-character repertoire
 /// `a-zA-Z0-9_`.
+#[repr(C)]
 #[derive(Copy, Clone)]
-pub struct Symbol(RawVal);
+pub struct Symbol(u128);
```

Comment:
We should be able to derive `Eq` on `Symbol` now, which will mean we can remove some hackery to get around not being able to use them in `match` statements.